portfolio/src/main/java/com/google/sps/comments/NewCommentServlet.java

```diff
@@ -61,8 +61,27 @@ private void addNewComment(HttpServletRequest request) {
     commentEntity.setProperty("message", commentMessage);
     commentEntity.setProperty("votes", 0);
 
+    // Determine the sentiment score of the comment
+    double sentimentScore = sentimentAnalyzer(commentMessage);
+
+    commentEntity.setProperty("sentimentScore", sentimentScore);
+
     // Put entity into datastore.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
   }
+
+  // Analyze the overall sentiment of the comment message
+  private double sentimentAnalyzer(String commentMessage) throws IOException {
```

Comment:
I think a more accurate name for this function could be "getSentimentScore" or "calculateSentimentScore".

Reply:
Thanks for the recommendation!